contracts/account/library.cairo

```diff
@@ -0,0 +1,558 @@
+    let (signer) = _signer.read();
+    with_attr error_message("argent: account not initialized") {
+        assert_not_zero(signer);
+    }
+    return ();
+}
+
+func assert_non_reentrant{syscall_ptr: felt*}() -> () {
+    let (caller) = get_caller_address();
+    with_attr error_message("argent: no reentrant call") {
+        assert caller = 0;
+    }
+    return ();
+}
+
+func assert_correct_version{syscall_ptr: felt*}(version: felt) -> () {
+    with_attr error_message("argent: invalid tx version") {
+        assert version = 1;
+    }
+    return ();
+}
+
+func assert_guardian_set{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
+    let (guardian) = _guardian.read();
+    with_attr error_message("argent: guardian required") {
+        assert_not_zero(guardian);
+    }
+    return ();
+}
+
+func assert_no_self_call(self: felt, call_array_len: felt, call_array: CallArray*) {
+    if (call_array_len == 0) {
+        return ();
+    }
+    assert_not_zero(call_array[0].to - self);
+    assert_no_self_call(self, call_array_len - 1, call_array + CallArray.SIZE);
+    return ();
+}
+
+// @notice Executes a list of contract calls recursively.
+// @param calls_len The number of calls to execute
+// @param calls A pointer to the first call to execute
+// @param response The array of felt to pupulate with the returned data
+// @return response_len The size of the returned data
+func execute_list{syscall_ptr: felt*}(calls_len: felt, calls: Call*, reponse: felt*, index: felt) -> (
+    response_len: felt
+) {
+    alloc_locals;
+
+    // if no more calls
+    if (calls_len == 0) {
+        return (0,);
+    }
+
+    // do the current call
+    let this_call: Call = [calls];
+    with_attr error_message("argent: multicall {index} failed") {
+        let res = call_contract(
+            contract_address=this_call.to,
+            function_selector=this_call.selector,
+            calldata_size=this_call.calldata_len,
+            calldata=this_call.calldata,
+        );
+    }
+    // copy the result in response
+    memcpy(reponse, res.retdata, res.retdata_size);
+    // do the next calls recursively
+    let (response_len) = execute_list(calls_len - 1, calls + Call.SIZE, reponse + res.retdata_size, index + 1);
+    return (response_len + res.retdata_size,);
+}
+
+func from_call_array_to_call{syscall_ptr: felt*}(
+    call_array_len: felt, call_array: CallArray*, calldata: felt*, calls: Call*
+) {
+    // if no more calls
+    if (call_array_len == 0) {
+        return ();
+    }
+
+    // parse the current call
+    assert [calls] = Call(
+        to=[call_array].to,
+        selector=[call_array].selector,
+        calldata_len=[call_array].data_len,
+        calldata=calldata + [call_array].data_offset
+        );
+
+    // parse the remaining calls recursively
+    from_call_array_to_call(
+        call_array_len - 1, call_array + CallArray.SIZE, calldata, calls + Call.SIZE
+    );
+    return ();
+}
+
+namespace ArgentModel {
```

Comment:
I don't like `ArgentRecovery` as it is too restrictive.

An alternative is to call it `ArgentAccount` since it is the logic specific to the Argent account, but then I would need to call the `ArgentAccount.cairo` file differently... probably something like `ArgentDefaultAccount.cairo` which you asked me to change in the previous PR :-)